src/PaperG/FirehoundBlob/Facebook/FacebookCreative.php

```diff
@@ -58,7 +63,9 @@ public function getObjects()
      */
     public function setType($type)
     {
-        $this->type = $type;
+        if (in_array($type, self::$validAdTypes)) {
+            $this->type = $type;    
+        }
```

Comment:
Alternatively, why not allow all values, and then check the types on `isValid()`?  I worry that this could lead to somewhat arcane errors.